analysis/quantifier-comparisons/03-memory_time_comparison.Rmd

```diff
@@ -58,66 +60,82 @@ For each run, we asked nextflow to ouput a trace.txt txt file using the
 -with-trace flag. These [trace files](https://www.nextflow.io/docs/latest/tracing.html) 
 include important information about each process such as run time and memory. 
 
-## Alevin
+Let's make a function to read in the trace files for each of the tools. 
+Each tool and configuration has it's own working directory with three trace.txt files: one containing the run for scRNA samples, one for the snRNA samples with the mRNA index, and one for the snRNA samples with the pre-mRNA index. 
+```{r}
+make_trace_df <- function(workdir, file_ext = ".trace.txt"){
+  setwd(workdir)
+  workdir_files <- list.files(workdir)
+  workdir_list <- lapply(workdir_files, function(x) readr::read_tsv(x))
+  # remove extension from trace file names
+  names(workdir_list) <- gsub(file_ext, "", workdir_files)
+  # create list of data frames for all trace files present in directory 
+  workdir_df <- bind_rows(workdir_list)
+}
+```
+
+## Alevin 
 
 ```{r}
-# read in alevin trace files
-alevin_trace <- readr::read_tsv(file.path(alevin_dir,
-                                          "alevin.trace.allsamples.txt"))
-alevin_trace$Tool = "Alevin"
-alevin_trace
+alevin_trace_df <- make_trace_df(alevin_dir, ".trace.txt")
+alevin_trace_df$Tool <- "Alevin"
 ```
+
 ## Alevin Fry 
 
 ```{r}
-alevin_fry_trace <- readr::read_tsv(
-  file.path(alevin_fry_dir, "alevin-fry.trace.allsamples.txt")
-)
+alevin_fry_trace_df <- make_trace_df(alevin_fry_dir, ".trace.txt")
+alevin_fry_trace_df$Tool <- "Alevin-Fry"
```

Comment:
Because for some reason past me thought this was made sense which it most definitely does not. It is now included in the function which is much simpler!